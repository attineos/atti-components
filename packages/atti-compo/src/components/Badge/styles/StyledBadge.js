import styled, { css } from 'styled-components'
import BigText from '../../BigText'

const borders = css`
  border-radius: ${({ theme }) => theme.components.badge.borders.radius};
`

const spaces = css`
  padding: ${({ theme }) =>
    `${theme.components.badge.spaces.paddingHeight} ${theme.components.badge.spaces.paddingWidth}`};
`

const colors = css`
  background-color: ${({ theme, type }) => theme.components.badge.colors.background[type]};
  color: ${({ theme }) => theme.components.badge.colors.text};
`

const fonts = css`
  line-height: ${({ theme }) => theme.components.badge.fonts.lineHeight};
  font-size: ${({ theme }) => theme.components.badge.fonts.fontSize};
  font-weight: ${({ theme }) => theme.components.badge.fonts.fontWeight};
`

const StyledBadge = styled(BigText.withComponent('span'))`
  ${borders}
  ${spaces}
  ${colors}
  ${fonts}
  display: ${({ theme }) => theme.components.badge.display};
`

export default StyledBadge
