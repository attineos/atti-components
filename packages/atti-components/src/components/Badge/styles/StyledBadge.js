import styled, { css } from 'styled-components'

const borders = css`
  border-radius: ${({ theme }) => theme.components.badge.borders.radius};
`

const spaces = css`
  padding: ${({ theme }) =>
    `${theme.components.badge.spaces.paddingHeight} ${theme.components.badge.spaces.paddingWidth}`};
`

const colors = css`
  background-color: ${({ theme, variance }) => theme.components.badge.colors.background[variance]};
  color: ${({ theme }) => theme.components.badge.colors.text};
`

const fonts = css`
  line-height: ${({ theme }) => theme.components.badge.fonts.lineHeight};
  font-size: ${({ theme }) => theme.components.badge.fonts.fontSize};
  font-weight: ${({ theme }) => theme.components.badge.fonts.fontWeight};
  font-family: ${({ theme }) => theme.components.badge.fonts.fontFamily};
  letter-spacing: ${({ theme }) => theme.components.badge.fonts.letterSpacing};
`

const StyledBadge = styled.span`
  ${borders}
  ${spaces}
  ${colors}
  ${fonts}
  display: inline-flex;
  align-items: center;
  justify-content: center;
  text-transform: uppercase;
`

export default StyledBadge
