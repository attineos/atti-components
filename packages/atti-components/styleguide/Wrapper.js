import React from 'react'

import ThemeProvider from '../src/components/ThemeProvider'
import { defaultTheme } from '../src/themes'

export default ({ children }) => <ThemeProvider otherTheme={defaultTheme}>{children}</ThemeProvider>
