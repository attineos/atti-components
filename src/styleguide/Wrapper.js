import React from 'react'
import PropTypes from 'prop-types'
import { remove, get } from 'lodash'

import Cookies from './helpers/Cookies'
import themes from './config/themes'

import ThemeProvider from '../components/ThemeProvider'

const updateWrappers = []

export const getWrappersToUpdate = () => updateWrappers

class Wrapper extends React.Component {
  static propTypes = {
    /**
     * The child node.
     */
    children: PropTypes.node.isRequired,
  }

  state = {
    selectedTheme: null,
  }

  componentWillMount() {
    updateWrappers.push(this)

    const selectedTheme = Cookies.hasItem('selectedTheme')
      ? Cookies.getItem('selectedTheme')
      : 'default'

    this.state.selectedTheme = get(themes, selectedTheme, null)
  }

  componentWillUnmount() {
    remove(updateWrappers, item => item === this)
  }

  handleSetTheme = newTheme =>
    this.setState(
      {
        selectedTheme: newTheme,
      },
      () => this.forceUpdate(),
    )

  render() {
    const { children } = this.props
    const { selectedTheme } = this.state

    return (
      <ThemeProvider theme={selectedTheme} key={Math.random()}>
        {children}
      </ThemeProvider>
    )
  }
}

export default Wrapper
