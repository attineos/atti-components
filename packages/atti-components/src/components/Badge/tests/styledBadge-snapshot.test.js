/* This file was generated automatically, please do not write in it. */
import React from 'react';
import StyledBadge from '../styles/StyledBadge';
import Wrapper from 'styleguide/Wrapper'
import renderer from 'react-test-renderer';

const mock = () => {}

it('StyledBadge renders correctly', () => {
  const tree = renderer
    .create(
      <Wrapper>
        <StyledBadge ></StyledBadge>
      </Wrapper>
      )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
    