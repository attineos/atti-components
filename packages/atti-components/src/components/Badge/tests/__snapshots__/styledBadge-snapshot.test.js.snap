// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledBadge renders correctly 1`] = `
<span
  className="atti-box"
>
  <span
    className="atti-box"
  >
    <span
      className="StyledBadge-sc-14ha03a-0 hKrCPP"
    />
  </span>
</span>
`;
